core/trino-main/src/main/java/io/trino/operator/aggregation/AggregationFromAnnotationsParser.java

```diff
@@ -127,20 +133,25 @@ private static AggregationHeader parseHeader(AnnotatedElement aggregationDefinit
     {
         AggregationFunction aggregationAnnotation = aggregationDefinition.getAnnotation(AggregationFunction.class);
 
-        return getNames(toParse, aggregationAnnotation).stream()
-                .map(name ->
+        return getNames(toParse, aggregationAnnotation).entrySet().stream()
+                .map(entry ->
                         new AggregationHeader(
-                                name,
+                                entry.getKey(),
+                                entry.getValue(),
                                 parseDescription(aggregationDefinition, toParse),
                                 aggregationAnnotation.decomposable(),
                                 aggregationAnnotation.isOrderSensitive(),
                                 aggregationAnnotation.hidden()))
                 .collect(toImmutableList());
     }
 
-    private static List<String> getNames(@Nullable AnnotatedElement outputFunction, AggregationFunction aggregationAnnotation)
+    private static Map<String, String> getNames(@Nullable AnnotatedElement outputFunction, AggregationFunction aggregationAnnotation)
```

Comment:
The Map here is actually a pair of `(actual name, [actual name + aliases])` in disguise.
It's not incidental that all map values are always the same.
Ideally we should represent this with a dedicated value class (defined here, privately).
I am OK leaving this as is for now, but please add a comment explaining the returned Map structure, so that it doesn't surprise a reader in the future.

Reply:
> Ideally we should represent this with a dedicated value class (defined here, privately).

Copy that.